fix(datastore): fix invalid model id field name implication

HuiSF opened the pull request:
*Issue #, if available:*

https://github.com/aws-amplify/amplify-flutter/issues/1565

*Description of changes:*

Add logic in Android implement removing `<modelName>` from `<modelName>.id` when creating native query predicate.

* The same logic exists in [iOS implementation](https://github.com/aws-amplify/amplify-flutter/blob/5c6d8eab63c1ea11e63d1ec8e82a975a4dffeb28/packages/amplify_datastore/ios/Classes/types/query/QueryPredicateBuilder.swift#L23-L29)
* The field name `<modelName>.id` for a model id file is redundant and unnecessary and breaks query predicate usage outside of SQLite in Android 
* This issue didn't break `DataStore.query` because Android generated SQL command happened to compatible with `<modelName>.id`

**NOTE:** This PR resolves the issue identified in Android, and ensures backwards compatibility for model generated wrongly.

PR to amplify-codegen to correct this behavior: https://github.com/aws-amplify/amplify-codegen/pull/429

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

dnys1 commented on packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/query/QueryPredicateBuilder.kt:
```diff
@@ -144,5 +144,14 @@ class QueryPredicateBuilder {
         }
+
+        @JvmStatic
+        fun extractIdFieldName(fieldName: String): String {
```
Is this just to be backwards compatible with old codegen? The codegen fix alone should work, right?

If it's to be backwards compatible? Should we have a test which verifies this old behavior?

HuiSF replied:
Yes. This is for backwards compatibility.
I'll add a test to assert the original behavior.

HuiSF commented:
Update:

* Removed extracting id field name logics from both iOS and Android implementation
* Implemented the backwards compatibility guard in flutter implementation
